tests/func/test_arithmetic.cpp

```diff
@@ -328,10 +328,10 @@ TEST(Arithmetic, OptionArrayLHSInt32)
   EXPECT_ARRAY_EQ(nd::is_na(data * 2), expected);
   EXPECT_ARRAY_EQ(nd::is_na(data / 1), expected);
 
-  auto add = nd::array{1, 41, 2}.ucast(ndt::type("?int32")).eval();
-  auto sub = nd::array{-1, 39, 0}.ucast(ndt::type("?int32")).eval();
-  auto mul = nd::array{0, 80, 2}.ucast(ndt::type("?int32")).eval();
-  auto div = nd::array{0, 40, 1}.ucast(ndt::type("?int32")).eval();
+  nd::array add = nd::empty(ndt::type("3 * ?int32")).assign({1, 41, 2});
+  nd::array sub = nd::empty(ndt::type("3 * ?int32")).assign({-1, 39, 0});
+  nd::array mul = nd::empty(ndt::type("3 * ?int32")).assign({0, 80, 2});
+  nd::array div = nd::empty(ndt::type("3 * ?int32")).assign({0, 40, 1});
```

Comment:
Is it be reasonable to make `nd::array({0, 40, 1}, "?int32")` or `nd::array({0, 40, 1}, "3 * ?int32")` work? A succinct spelling of this would be nice.